services/airflow/dags/helpers/test_helpers/helpers.py

```diff
@@ -84,9 +86,12 @@ def if_var_exists_in_dag_conf_use_as_first_arg(
 
 def create_task_instance(dag_id: str, task_id: str, url: str = None):
     dag = DagBag().get_dag(dag_id)
-    execution_date = datetime.now()
+    now = datetime.now()
     dr = dag.create_dagrun(
-        state="running", run_type=DagRunType("manual"), execution_date=execution_date
+        state="running",
+        run_type=DagRunType("manual"),
+        execution_date=now,
+        data_interval=(days_ago(3), days_ago(0)),
```

Comment:
what does this interval do?

Reply:
tldr: It makes deprecation warnings during tests go away.

This is undocumented but at the same time it is deprecated not using it.
From the source code I take it that this is used to limit the time interval for which the dag is executed in case of backfill runs (e.g. schedule tasks that are executed now but for a dates in time window that is in the past)